docs/adr/adr_003_bundling_collection_data.md

```diff
@@ -0,0 +1,209 @@
+- Implementation should be resilient to race conditions on attachment downloads. Unpacking the bundle can overwrite attachments that had been previously downloaded individually, and individual downloads can overwrite unbundled ones
+- Bundle format should be easily readable by client (third-party libraries are hard to ship in Gecko)
+
+References
+- [The intermediates download code](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/security/manager/ssl/RemoteSecuritySettings.sys.mjs#335-365) that fetches attachments in bulk, and stores state in records
+- [Download of crlite files](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/security/manager/ssl/RemoteSecuritySettings.sys.mjs#620-637)
+- [Download of all regions files](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/security/manager/ssl/RemoteSecuritySettings.sys.mjs#335-365)
+- [Download of all addons suggestions files](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/browser/components/urlbar/private/AddonSuggestions.sys.mjs#87-101)
+
+### Metrics To Measure Before/After
+We want to measure CPU usage, total bandwidth used, duration of syncronization on fresh profiles, and battery usage for full synchronization. We should be able to gather most of this by running power and performance profiles on different hardware.
+
+Hardware to include:
+- Mac Desktop/Laptop
+- Linux Desktop/Laptop
+- Windows Desktop/Laptop
+- Android Mobile
+- iOS Mobile (does it use remote-settings?)
+
+
+## Non Goals
+- Support diffs or delta bundles for clients that already have local data
+- Reduce client performance (ex: we shouldn't increase power usage for clients)
+
+## Assumptions
+- We will find a way for Gecko clients to extract at least one archive format (very likely using third-party library)
+
+## Decision Drivers
+- Keep costs low
+    - With millions of clients, the cost of shipping data is always a concern
+- Server complexity & performance
+    -Monitoring and operating should remain straightforward (ex: limit tech and deployment complexity if possible)
+    -Performance of the service should not be affected
+    - The code in place to support this feature should not involve complex components, libraries, or architecture (eg. to support resumable downloads and range requests)
+- Long-term Maintenance
+- Clients complexity
+    - How complicated it is to know which bundle URL to pull
+
+## Considered Options
+1. Build bundles on attachment upload/removal
+2. Build bundles on Cloud storage event
+3. Build bundles on RS changes approval
+4. Add new API endpoint to build or serve last bundle
+5. Build bundles on schedule
+
+## Decision Outcome
+Chosen option: [Option 5](#Option-5) because it is easy to reason about, trivial to implement, does not affect user experience, and would not increase the overall service complexity. 
```

Comment:
One thing that isn't immediately clear (to me) about some of the options and the resulting decision is how we weigh some pros and cons against each other.

For example, we call option 5 "trivial" to implement, but there are some aspects of it that make it seem less-trivial than Option 1. But we're willing to take on some non-trivialness for the performance / UX improvements we'd gain, which makes sense. Maybe we can expand this decision outcome to mention that.